src/utils/common.utils.ts

```diff
@@ -59,3 +59,35 @@ export function uuidV4() {
         }
     );
 }
+
+export const formatDate = (date: Date) => {
+    const months = [
+        'Jan',
+        'Feb',
+        'Mar',
+        'Apr',
+        'May',
+        'Jun',
+        'Jul',
+        'Aug',
+        'Sep',
+        'Oct',
+        'Nov',
+        'Dec',
+    ];
+
+    const day = date.getDate().toString().padStart(2, '0'); // Add leading zero if needed
+    const month = months[date.getMonth()];
+    const year = date.getFullYear();
+
+    let hour = date.getHours();
+    const minute = date.getMinutes().toString().padStart(2, '0'); // Add leading zero if needed
+    const ampm = hour >= 12 ? 'PM' : 'AM';
+
+    if (hour > 12) {
+        hour -= 12;
+    }
+
+    const formattedDate = `${day} ${month}, ${year} ${hour}:${minute} ${ampm}`;
+    return formattedDate;
+};
```

Comment:
You need to import `format` from `date-fns` package.

```suggestion
export const formatDate = (date: string | Date, showTime?: boolean) => {
    return format(
        GetDateValue(date),
        showTime ? 'dd MMM, yyyy hh:mm a'  :  'dd MMM, yyyy'
    );
};
```

```
export const GetDateValue = (date: any, format?: string) => {
    if (!date || isDate(date)) return date;
    if (format) {
        return parse(date, format, new Date());
    }
    return new Date(date);
};
```